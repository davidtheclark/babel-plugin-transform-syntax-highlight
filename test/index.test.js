'use strict';

const babel = require('babel-core');
const babelEs2015 = require('babel-preset-es2015');
const babelReact = require('babel-preset-react');
const plugin = require('..');

const transform = (code, options) => {
  const pluginValue = options ? [plugin, options] : plugin;
  return babel.transform(code, {
    presets: [babelEs2015, babelReact],
    plugins: [pluginValue]
  }).code;
};

test('basic html', () => {
  const code = `
    const highlight = require('babel-plugin-transform-syntax-highlight/highlight');
    const snippet = highlight.html({ language: 'javascript' }, \`
      const foo = 'bar';
      const bar = '{# props.bar #}';
    \`);
  `;
  expect(transform(code)).toMatchSnapshot();
});

test('basic react', () => {
  const code = `
    const highlight = require('babel-plugin-transform-syntax-highlight/highlight');
    const snippet = highlight.react({ language: 'javascript' }, \`
      const foo = 'bar';
      const bar = '{# props.bar #}';
    \`);
  `;
  expect(transform(code)).toMatchSnapshot();
});

test('import instead of require', () => {
  const code = `
    import hl from 'babel-plugin-transform-syntax-highlight/highlight';
    const snippet = hl.html({ language: 'javascript' }, \`
      const foo = 'bar';
      const bar = '{# props.bar #}';
    \`);
  `;
  expect(transform(code)).toMatchSnapshot();
});

test('require in nested scope', () => {
  const code = `
    function x() {
      if (true) {
        const h = require('babel-plugin-transform-syntax-highlight/highlight');
        const snippet = h.html({ language: 'javascript' }, \`
          const bar = x => x * {# props.factor #};
        \`);
      }
    }
  `;
  expect(transform(code)).toMatchSnapshot();
});

test('code as string literal', () => {
  const code = `
    const highlight = require('babel-plugin-transform-syntax-highlight/highlight');
    const snippet = highlight.react({ language: 'javascript' }, 'var x = 3;');
  `;
  expect(transform(code)).toMatchSnapshot();
});

test('Babel options', () => {
  const options = {
    langauge: 'javascript',
    packageName: 'highlighter',
    delimiters: ['{{', '}}']
  };
  const code = `
  import hl from 'highlighter';
  const snippet = hl.html(\`
    const foo = {{ props.foo }};
    const bar = '{# props.bar #}';
  \`);
  `;
  expect(transform(code, options)).toMatchSnapshot();
});

test('Local options', () => {
  const code = `
    const h = require('babel-plugin-transform-syntax-highlight/highlight');
    const html = h.html({
      language: 'javascript',
      delimiters: ['[[', ']]']
    }, \`const bar = '[[ props.bar ]]';\`);
    const element = h.react({
      language: 'javascript',
      highlight: 'prism',
      delimiters: ['{{', '}}']
    }, \`const bar = '{{ props.bar }}';\`);
  `;
  expect(transform(code)).toMatchSnapshot();
});
