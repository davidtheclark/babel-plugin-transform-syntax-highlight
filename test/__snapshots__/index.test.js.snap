// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel options 1`] = `
"'use strict';

var snippet = function snippet(props) {
  return '<pre><code class=\\"hljs\\"><span class=\\"hljs-attribute\\">const foo</span> = ' + props.foo + ';\\\\n<span class=\\"hljs-attribute\\">const bar</span> = <span class=\\"hljs-string\\">\\\\'{# props.bar #}\\\\'</span>;</code></pre>';
};"
`;

exports[`Local options 1`] = `
"'use strict';

var html = function html(props) {
  return '<pre><code class=\\"hljs\\"><span class=\\"hljs-keyword\\">const</span> bar = <span class=\\"hljs-string\\">\\\\'' + props.bar + '\\\\'</span>;</code></pre>';
};
var element = function element(props) {
  return React.createElement(
    'pre',
    null,
    React.createElement(
      'code',
      { className: 'language-javascript' },
      React.createElement(
        'span',
        { className: 'token keyword' },
        'const'
      ),
      ' bar ',
      React.createElement(
        'span',
        { className: 'token operator' },
        '='
      ),
      ' ',
      React.createElement(
        'span',
        { className: 'token string' },
        '\\\\'',
        props.bar,
        '\\\\''
      ),
      React.createElement(
        'span',
        { className: 'token punctuation' },
        ';'
      )
    )
  );
};"
`;

exports[`basic html 1`] = `
"'use strict';

var snippet = function snippet(props) {
  return '<pre><code class=\\"hljs\\"><span class=\\"hljs-keyword\\">const</span> foo = <span class=\\"hljs-string\\">\\\\'bar\\\\'</span>;\\\\n<span class=\\"hljs-keyword\\">const</span> bar = <span class=\\"hljs-string\\">\\\\'' + props.bar + '\\\\'</span>;</code></pre>';
};"
`;

exports[`basic react 1`] = `
"'use strict';

var snippet = function snippet(props) {
  return React.createElement(
    'pre',
    null,
    React.createElement(
      'code',
      { className: 'hljs' },
      React.createElement(
        'span',
        { className: 'hljs-keyword' },
        'const'
      ),
      ' foo = ',
      React.createElement(
        'span',
        { className: 'hljs-string' },
        '\\\\'bar\\\\''
      ),
      ';',
      \\"\\\\n\\",
      React.createElement(
        'span',
        { className: 'hljs-keyword' },
        'const'
      ),
      ' bar = ',
      React.createElement(
        'span',
        { className: 'hljs-string' },
        '\\\\'',
        props.bar,
        '\\\\''
      ),
      ';'
    )
  );
};"
`;

exports[`code as string literal 1`] = `
"'use strict';

var snippet = function snippet(props) {
    return React.createElement(
        'pre',
        null,
        React.createElement(
            'code',
            { className: 'hljs' },
            React.createElement(
                'span',
                { className: 'hljs-keyword' },
                'var'
            ),
            ' x = ',
            React.createElement(
                'span',
                { className: 'hljs-number' },
                '3'
            ),
            ';'
        )
    );
};"
`;

exports[`highlighting numeric values 1`] = `
"'use strict';

var snippet = function snippet(props) {
  return '<pre><code class=\\"hljs\\"><span class=\\"hljs-keyword\\">const</span> foo = <span class=\\"hljs-string\\">\\\\'' + props.foo + '\\\\'</span>;\\\\n<span class=\\"hljs-keyword\\">const</span> bar = <span class=\\"hljs-number\\">' + props.bar + '</span>;</code></pre>';
};"
`;

exports[`import instead of require 1`] = `
"'use strict';

var snippet = function snippet(props) {
  return '<pre><code class=\\"hljs\\"><span class=\\"hljs-keyword\\">const</span> foo = <span class=\\"hljs-string\\">\\\\'bar\\\\'</span>;\\\\n<span class=\\"hljs-keyword\\">const</span> bar = <span class=\\"hljs-string\\">\\\\'' + props.bar + '\\\\'</span>;</code></pre>';
};"
`;

exports[`require in nested scope 1`] = `
"'use strict';

function x() {
  if (true) {
    var snippet = function snippet(props) {
      return '<pre><code class=\\"hljs\\"><span class=\\"hljs-keyword\\">const</span> bar = <span class=\\"hljs-function\\"><span class=\\"hljs-params\\">x</span> =&gt;</span> x * <span class=\\"hljs-number\\">' + props.factor + '</span>;</code></pre>';
    };
  }
}"
`;
